diff --git a/crates/vpg/src/variability_zielonka.rs b/crates/vpg/src/variability_zielonka.rs
index 0cd530b1..67936caf 100644
--- a/crates/vpg/src/variability_zielonka.rs
+++ b/crates/vpg/src/variability_zielonka.rs
@@ -119,28 +119,10 @@ pub fn solve_variability_product_zielonka<'a>(
     project_variability_parity_games_iter(vpg, timing).map(|result| {
         match result {
             Ok((Projected { bits, bdd, game }, timing)) => {
-                let (reachable_pg, projection) = timing.measure("reachable", || compute_reachable(&game));
-
                 debug!("Solving projection on {}...", FormatConfig(&bits));
 
-                let (pg_solution, _) = solve_zielonka(&reachable_pg, false);
-                let mut new_solution = [
-                    bitvec![usize, Lsb0; 0; vpg.num_of_vertices()],
-                    bitvec![usize, Lsb0; 0; vpg.num_of_vertices()],
-                ];
-                for v in game.iter_vertices() {
-                    if let Some(proj_v) = projection[*v] {
-                        // Vertex is reachable in the projection, set its solution
-                        if pg_solution[0][proj_v] {
-                            new_solution[0].set(*v, true);
-                        }
-                        if pg_solution[1][proj_v] {
-                            new_solution[1].set(*v, true);
-                        }
-                    }
-                }
-
-                Ok((bits, bdd, new_solution))
+                let (pg_solution, _) = solve_zielonka(&game, false);
+                Ok((bits, bdd, pg_solution))
             }
             Err(result) => Err(result),
         }
